### ISSUE TITLE
fish for ZNC (feature request) 

### ISSUE THREAD

**Issue by antonstarlid** [jreese/znc-push#150]
Hay i use the fish for ZNC and the awsome push module, the only problem is that if someone send me a massage where its used fish i can only see the fish code and not the massage.

I believe this is because the massage is send before the fish module can decrypt it, or its just can´t be read by the push module as the fish and push are two different modules.

I just wonder if its anyway around this, and thanks for an awesome module!

\Anton

**Comment by Mikaela** [comment]
I think this can be fixed by editing znc.conf (NOT RECOMMENDED) and putting the line to load push before the line that loads fish.

**Comment by antonstarlid** [comment]
The push module is loaded in the user in the ZNC while the fish is loaded inside every network.

So im not sure how you mean Mikaela

\Anton

**Comment by Mikaela** [comment]
Oh, I don't know in that case, but maybe someone else has some kind of idea.

**Comment by antonstarlid** [comment]
Mikaela

I think that push module and fish module work for them self, It means that the push module has to now somehow that it is a decrypted fish massage it get and decrypt it. Else it will never work

Think only way to done this is to add code to only the push module or both fish and push

**Comment by jreese** [comment]
I don't think this is going to be feasible, unfortunately.  But if someone with better knowledge of the fish module can submit a pull request with code to support it, I'd be more than happy to review it.

**Comment by antonstarlid** [comment]
How can´t it be feasible? The only thing thats need is to make znc-push to know that it is an crypted massage and read it, i don´t know if its need to be modify both fish and znc-push or just znc-push.

But i have really hard to believe that its hard to fix, znc fish can be find here http://pastebin.com/K8zvJ89v

What make you think thats its feasible jreese?

\ Anton

**Comment by jreese** [comment]
Not feasible, because:

- this will require a dependency on the code from the fish module at the very least, because there's no way for one module to communicate with another.  This will make the build more complicated, which I don't like.
- this puts the burden of decryption on znc-push, which means znc-push then needs to know what/where the decryption keys are, which means more configuration options, or reaching into the configuration of another module, which isn't support by ZNC.
- this is not forward compatible with my goals of moving away from c++ to python in the near future.

Note that I'm not saying it's impossible, just that the effort to implement it is more than the benefit IMO, and requires implementing it in ways that I don't like/want.  Again, if someone else is willing to put forth that effort, I'll gladly review the code, but depending on the implementation, I might not accept it into master, for the reasons I mentioned above.

**Comment by Mikaela** [comment]
> * this will require a dependency on the code from the fish module at the very least, because there's no way for one module to communicate with another. This will make the build more complicated, which I don't like.
> * this puts the burden of decryption on znc-push, which means znc-push then needs to know what/where the decryption keys are, which means more configuration options, or reaching into the configuration of another module, which isn't support by ZNC.

depends on https://github.com/znc/znc/issues/608

**Comment by jreese** [comment]
Yes, if that ever becomes a thing, then compatibility with fish is certainly something I would look into, but it would depend on fish itself having an appropriate interface for allowing modules like znc-push to use it.
